server/src/main/java/org/apache/cassandra/sidecar/coordination/ElectorateMembership.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.sidecar.coordination;
+
+/**
+ * An interface that defines if the local Sidecar instance should participate during the choosing of a single
+ * instance executor process.
+ */
+public interface ElectorateMembership
+{
+    /**
+     * @return {@code true} if the local Sidecar instance should participate in the process,
+     * {@code false} otherwise
+     */
+    boolean shouldParticipate();
```

Comment:
I was struggling with the name of this class/method for some reason... I think the method should be called `isMember` rather than `shouldParticipate` and it all makes much more sense. Comments should also reflect that this (and the TokenZeroElectorateMembership class) are just about membership in a particular electorate (which I'm still not sure of name-wise, but I haven't come up with anything better yet either).